.github/workflows/docs.yml

```diff
@@ -0,0 +1,53 @@
+# https://github.com/mitmproxy/pdoc/blob/main/.github/workflows/docs.yml
+
+name: PyHa Documentation
+
+# build the documentation whenever there are new commits on main
+on:
+  push:
+    branches:
+      - main
+      - np-generate-docs
+
+# security: restrict permissions for CI jobs.
+permissions:
+  contents: read
+
+jobs:
+  # Build the documentation and upload the static HTML files as an artifact.
+  build:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip poetry
+          poetry install
+
+      - name: Build Documentation
+        run: poetry run python -m pdoc ./PyHa --docformat google -o docs/
+        env:
+          PDOC_ALLOW_EXEC: 1
```

Comment:
It might be good if we also add a status badge indicating quality of documentation, especially if the end goal is for PyHa to be a package others can use.